.github/workflows/mmctl-release.yml

```diff
@@ -0,0 +1,137 @@
+name: mmctl release
+on:
+  # This workflow gets triggered from the Argo Events platform.
+  # Check the following repo for details: https://github.com/mattermost/delivery-platform
+  workflow_dispatch:
+    inputs:
+      payload:
+        description: "The triggerer payload"
+        required: true
+        type: string
+      dry_run:
+        type: string # True if left empty. Set to "false" to enable the S3 upload
+        required: false
+
+defaults:
+  run:
+    shell: bash
+    working-directory: server
+
+env:
+  go-version: "1.19.5"
+
+jobs:
+  update-initial-status:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: mattermost/actions/delivery/update-commit-status@main
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+        with:
+          repository_full_name: ${{ github.repository }}
+          commit_sha: ${{ github.sha }}
+          context: mmctl/publish
+          description: Publish mmctl binaries
+          status: pending
+
+  expose-incoming-variables:
+    runs-on: ubuntu-22.04
+    outputs:
+      ref_name: ${{ env.TRIGGERER_BRANCH }}
+    steps:
+      - uses: mattermost/actions/delivery/expose-events-vars@a74f6d87f847326c04d326bf1908da40cb9b3556
+        with:
+          payload: ${{ inputs.payload }}
+
+  build:
+    runs-on: ubuntu-22.04
+    needs:
+      - update-initial-status
+      - expose-incoming-variables
+    strategy:
+      matrix:
+        os: [linux, darwin, windows]
+        arch: [amd64, arm64]
+        exclude:
+          - os: windows
+            arch: arm64
+    env:
+      GOBIN: "bin"
+      GOOS: "${{ matrix.os }}"
+      GOARCH: "${{ matrix.arch }}"
+      DRY_RUN: "${{ inputs.dry_run || true }}"
+    steps:
+      - name: ci/checkout-repo
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      - name: ci/setup-go
+        uses: actions/setup-go@4d34df0c2316fe8122ab82dc22947d607c0c91f9 # v4.0.0
+        with:
+          go-version: ${{ env.go-version }}
```

Comment:
Is there a reason for not using go.sum to get the go version? 
https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

Reply:
afaiu it's because there's no patch version in our `go.mod` file, so by specifying the go version this way we get a more consistent build environment over time (instead of using the latest available 1.19 version).

Although I just adhered to the standard used in the other workflow files here, so this is a more general question than this PR. Perhaps we can raise this in the Developers channel

Reply:
Ah I just noticed that your link also mentions that no patch version can be contained within `go.mod`, and that the latest patch version would be picked if we use that method.

It seems that the only way to pin an exact toolchain version is by using this approach